xmtp_proto/src/error.rs

```diff
@@ -0,0 +1,138 @@
+use openmls::prelude::tls_codec::Error as TlsCodecError;
```

Comment:
Is this `error.rs` file generated? Maybe I missed something, why do we need a new error type different from before?

Reply:
I just moved it to its own file as a drive-by refactor.

I added the InternalError and all of its subtypes to allow for error conversions.